src/java/org/apache/cassandra/index/sai/disk/v1/MergeOneDimPointValues.java

```diff
@@ -47,22 +47,9 @@
 
     public MergeOneDimPointValues(List<BKDReader.IteratorState> iterators, AbstractType termComparator) throws IOException
     {
-        queue = new MergeQueue(iterators.size());
-        this.scratch = new byte[TypeUtil.fixedSizeOf(termComparator)];
-        for (BKDReader.IteratorState iterator : iterators)
-        {
-            if (iterator.hasNext())
-            {
-                queue.add(iterator);
-            }
-            else
-            {
-                iterator.close();
-            }
-        }
+        this(iterators, TypeUtil.fixedSizeOf(termComparator));
     }
 
-    @VisibleForTesting
     public MergeOneDimPointValues(List<BKDReader.IteratorState> iterators, int bytesPerDim) throws IOException
     {
         queue = new MergeQueue(iterators.size());
```

Comment:
This is regarding the next line, I just can't comment on that line.

We don't need to allocate the `scratch` buffer because we don't appear to be using it. The only thing we use it for is it's length so we could just store the length and return that from `getBytesPerDimension()`

Reply:
ok, fixed